src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMigrationService.kt

```diff
@@ -0,0 +1,96 @@
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.PrisonPersonMigrationMappingRequest
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.PhysicalAttributesResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.PrisonerId
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.PrisonerPhysicalAttributesResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.AuditService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationHistoryService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationType
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.NomisApiService
+import java.time.LocalDateTime
+
+@Service
+class PrisonPersonMigrationService(
+  queueService: MigrationQueueService,
+  private val prisonPersonNomisApiService: PrisonPersonNomisApiService,
+  private val nomisService: NomisApiService,
+  private val prisonPersonMappingService: PrisonPersonMappingApiService,
+  private val prisonPersonDpsService: PrisonPersonDpsApiService,
+  migrationHistoryService: MigrationHistoryService,
+  telemetryClient: TelemetryClient,
+  auditService: AuditService,
+  @Value("\${page.size:1000}") pageSize: Long,
+  @Value("\${complete-check.delay-seconds}") completeCheckDelaySeconds: Int,
+  @Value("\${complete-check.count}") completeCheckCount: Int,
+) : MigrationService<PrisonPersonMigrationFilter, PrisonerId, PrisonerPhysicalAttributesResponse, PrisonPersonMigrationMappingRequest>(
+  queueService = queueService,
+  auditService = auditService,
+  migrationHistoryService = migrationHistoryService,
+  mappingService = prisonPersonMappingService,
+  telemetryClient = telemetryClient,
+  migrationType = MigrationType.PRISONPERSON,
+  pageSize = pageSize,
+  completeCheckDelaySeconds = completeCheckDelaySeconds,
+  completeCheckCount = completeCheckCount,
+) {
+  override suspend fun getIds(
+    migrationFilter: PrisonPersonMigrationFilter,
+    pageSize: Long,
+    pageNumber: Long,
+  ): PageImpl<PrisonerId> = nomisService.getPrisonerIds(
+    pageNumber = pageNumber,
+    pageSize = pageSize,
+  )
+
+  override suspend fun migrateNomisEntity(context: MigrationContext<PrisonerId>) {
+    log.info("attempting to migrate ${context.body}")
+    val offenderNo = context.body.offenderNo
+
+    val physicalAttributes = prisonPersonNomisApiService.getPhysicalAttributes(offenderNo)
+    val dpsIds = mutableListOf<Long>()
+    physicalAttributes.bookings.forEach { booking ->
+      booking.physicalAttributes.forEach { pa ->
+        val (lastModifiedAt, lastModifiedBy) = pa.lastModified()
+        prisonPersonDpsService.migratePhysicalAttributes(
```

Comment:
Not convinced this is a good idea, so for each prisoner you will be called DPS several times, what do you do if the sat the 6th call fails, won't it migrate the first 5 all over again and the mappings will be screwed up?